### PR TITLE
Working ristretto code added

### PR DESCRIPTION
**Pull request by janis-hag**
I have added the possibility to change the basefile as an argument to __init__ in cred1 in order to be able to use a different cfg. However by default in behaves as before.

### CODE REVIEW

File: camstack/cams/cred1.py [hunk 1]
```diff
@@ -202,6 +207,7 @@ def _get_cropping(self) -> Tuple[int, int, int, int]:
         logg.debug('_get_cropping @ CRED1')
         _, xx, yy = self.send_command('cropping raw').split(
                 ':')  # return is "(on|off):x0-x1:y0-y1"
+        # The line below will crash if cropping is set to a single column such that xx, yy = 10, 1-256
```
Comment (DasVinch) [inline camstack/cams/cred1.py:210]: Good point, the case never occured to me!

File: camstack/cams/cred1.py [hunk 2]
```diff
@@ -571,6 +577,30 @@ def _fill_keywords(self) -> None:
         self._set_formatted_keyword('GAIN', 1.98)
 
 
+class Ristretto(CRED1):
```
Comment (DasVinch) [inline camstack/cams/cred1.py:580]: Implementing 
```python
def _constructor_finalize(self) -> None:
    etc...
```
in your CRED1 subclass is now mandatory!
Check out the difference between Apapane and Iiwi.

**Comment by DasVinch** [comment]
Thanks for the PR. Most useful.
See my comment on _constructor_finalize, and then I'll merge.

**Comment by janis-hag** [comment]
Added  _constructor_finalize to the ristretto class

**Pull request merged**